Code review - homework #1

GitEsense opened the pull request.

GitEsense committed:
Создание дирректории для ДЗ №1

GitEsense committed:
Третья ступень - домашнее задание

GitEsense committed:
четвёртая ступень - домашнее задание

Семантическая вёрстка

GitEsense committed:
шестая ступень - домашнее задание

Добавлена папка 2-Form и обновлён .gitignore

GitEsense committed:
update .gitignore

GitEsense committed:
Небольшие корректировки в структуру

GitEsense committed:
Обновлён файл .gitignore

SergeyKrasnolobov commented on 1-html/1-Blog/index.html:
```diff
@@ -0,0 +1,94 @@
+<!DOCTYPE html>
+<html lang="en">
```
Почему lang en?

GitEsense replied:
Это пресет по умолчанию в VS Code - даже не трогал)

GitEsense replied:
Изменено, впредь буду обращать внимание (надо сделать уже себе готовый пресет)

SergeyKrasnolobov commented on 1-html/1-Blog/index.html:
```diff
@@ -0,0 +1,94 @@
+<html lang="en">
+    <head>
+        <meta charset="UTF-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
```
Мета тегов можно добавить, descr, author

GitEsense replied:
Это пресет по умолчанию в VS Code - даже не трогал)
Но как вариант

GitEsense replied:
Добавил, надеюсь верно)

SergeyKrasnolobov commented on 1-html/1-Blog/index.html:
```diff
@@ -0,0 +1,94 @@
+            <nav>
+                <ul>
+                    <li>
+                        <a href=""><span>B</span> Мой Блог</a>
```
Это относится к логитипу, а не к навигации, вынеси в отдельный див

GitEsense replied:
Блин, а ведь сначала так сделал, хотя разве это именно лого, как ссылка не должен быть как "главная" ?

SergeyKrasnolobov replied:
Так по клику на логотип сайта всегда перебрасывает на глоавную. Ты оставь его как ссылку, просто вынеси из блока с навигайцией

GitEsense replied:
Сделано

SergeyKrasnolobov commented on 1-html/1-Blog/index.html:
```diff
@@ -0,0 +1,94 @@
+                    <h1>
+                        Теория цвета - Как цвет влияет на поведение потребителей
+                    </h1>
+                    <div>
```
я бы сделал через article, так как этот блок может переиспользоваться в других частях сайта

GitEsense replied:
А Article вложить в Article имеет право на жизнь? или лучше вообще убрать? и сделать отдельно секциями?

SergeyKrasnolobov replied:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article

Дока не запрещает) Но смотри по семантике - section - блок для какого то контента, а article - это и есть контент.

```html
<section>
  <article>...</article>
  <article>...</article>
  <article>...</article>
</section>
```

Но опять же если ты считаешь что вместо seection подойдет article, то можешь юзать его. Семантика - вопрос всегда о двух стульях)

GitEsense replied:
Ну в моём случае было например 

```
<article>
   <section>...</section>
   <section>...</section>
   <section>...</section>
</article>
```

Я понимаю, что article я могу переиспользовать, тогда как section - это блок с доп.информацией или просто выделенная часть (читай div семантический), в то время как article я рассматривал как корневой div, копируя который я выношу всё, что хранилось в нём как переиспользование (статья блога \ новость)

Верное понимание?

SergeyKrasnolobov replied:
Ну в твоем случаем было не верно) а понимаешь вроде как верно

SergeyKrasnolobov commented on 1-html/1-Blog/index.html:
```diff
@@ -0,0 +1,94 @@
+                    <img
+                        src="./images/image1.png"
+                        alt="Изображение телевизора с подсветкой"
+                        style="
```
почему инлайн стилями?

GitEsense replied:
Там стиля вообще быть не должно было - это "пощупать" - первое дз сдавалось как "конструкция" html)
остальные буду заливать по порядку, когда в девелоп солью всё

SergeyKrasnolobov replied:
Ок)

SergeyKrasnolobov replied:
Я просто вижу что ты еще стилизацию добавил, по этому и спросил)

GitEsense replied:
Ну да, баловался, вообще в данном коммите надо бы убрать)

SergeyKrasnolobov commented on 1-html/1-Blog/index.html:
```diff
@@ -0,0 +1,94 @@
+                                lightgray 50% / cover no-repeat;
+                        "
+                    />
+                    <div>
```
зачем тут дивы? тег `p` имеет блочное поведение

GitEsense replied:
Когда со стилями баловался - закладывал туда, чтобы общие стили принимало - убрал за ненадобностью

SergeyKrasnolobov commented on 1-html/1-Blog/styles/archive.css:
```diff
@@ -0,0 +1,129 @@
+li > a > span {
```
А почему не классами сделал?

GitEsense replied:
Это делалось "специально" таким образом, потому что на момент, когда собирал - не проходил ещё уроки "селекторы классов" по CSS

css стили буду лить уже в ветку (2-css), согласно порядку сдачи ДЗ )

SergeyKrasnolobov commented on 1-html/1-Blog/styles/archive.css:
```diff
@@ -0,0 +1,129 @@
+article span {
+    opacity: 0.6;
+}
+article > div > p + span {
```
Так тоже не пойдет, чем длиннее у тебя логика поиска элемента, тем дольше происходит парсинг

GitEsense replied:
Это делалось "специально" таким образом, потому что на момент, когда собирал - не проходил ещё уроки "селекторы классов"  по CSS

SergeyKrasnolobov commented on 1-html/2-Form/index.html:
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
+<html lang="en">
```
Язык не тот

GitEsense replied:
Исправим - шаблонное значение через пресет VS Code

SergeyKrasnolobov commented on 1-html/2-Form/index.html:
```diff
@@ -0,0 +1,50 @@
+<html lang="en">
+    <head>
+        <meta charset="UTF-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
```
Мета теги

GitEsense replied:
Суть ДЗ была - построить конструкцию, стили прилипли исключительно случайно (когда щупал чисто для себя), по вопросам "семантики" и использования блоков - согласен)
Про мета теги даже близко пока особо речи не шло (о их видах, использовании, лишь о некоторых параметрах), поэтому и не трогал head вообще можно сказать)

SergeyKrasnolobov replied:
Ну смотри, я сторонник того чтобы делать чуть больше чем дано в задании, это же для тебя будет плюсом)

GitEsense replied:
Абсолютно согласен, про теги почитаю, благодарю)

GitEsense committed:
Внесены правки согласно комментариям

Убраны лишние стили, чтобы не отвлекать, добавлены мета теги, исправления в вёрстке (лишние дивы убраны)

SergeyKrasnolobov commented on 1-html/1-Blog/index.html:
```diff
@@ -1,18 +1,17 @@
     <body>
         <header>
             <nav>
+                <a href=""><span>B</span> Мой Блог</a>
```
Я бы наверное так предложил сделать)

```html
 <a href="">
  <div class='logo'>
     <span class='logo__img'>B</span>
     <span class = 'logo__text'>Мой Блог</span>
  </div>
</a>

```

GitEsense replied:
Я же правильно понимаю, что в данном случае div используется как обёртка и не более?
Я такую структуру рассматривал позднее (когда шла разбивка по классам), но соглашусь, лучше сразу так завернуть)

SergeyKrasnolobov replied:
Да да , все верно) Конечно лучше сразу заранее все продумывать чтобы меньше работы было потом

GitEsense replied:
Ну в общем сделано))
Ну а про классы - я уже говорил - это же самое первое ДЗ - которое подразумевало структуру, а вот второе ДЗ (ветка) по 2-css - там уже "марафет", оно так вышло, что я сейчас уже на анимациях, а сдаю ещё первую домашку)

В общем если всё ок, прошу замерджить и перейду к сдаче второй домашки)

SergeyKrasnolobov reviewed (commented):
По остальным пунктам все ок, только я бы поменял структуру для лого

GitEsense committed:
Изменена структура ссылки Лого(добавлен спан и обёртка, классы будут позднее)

GitEsense commented:
@SergeyKrasnolobov прошу отметить Код ревью-1 purpleschool как "выполнено", а так же хотелось бы узнать - есть ли альтернативный метод связи? (не discord, но телеграм например)
